src/main.rs

```diff
@@ -2,8 +2,15 @@
 // Licensed under the MIT License.
 
 use std::process::ExitCode;
+use std::sync::Arc;
+use std::time::Duration;
 
 use anyhow::Context;
+
+use azure_core::{ExponentialRetryOptions, RetryOptions};
+use azure_identity::AzureCliCredential;
+use azure_svc_imds::Client as ImdsClient;
```

Comment:
I looked at the azure sdk source quickly but did not see how to configure which http error codes to retry for. We need to ensure the following errors are retried: 403, 404, 410, 429, 500, 503.  While there are contention on whether the use of these error codes are appropriate on Azure, the fact remains that Azure platform will emit these errors and retrying is required.